showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.showcase.v1beta1.it;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.gax.core.GaxProperties;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.StatusCode.Code;
+import com.google.api.gax.tracing.ApiTracerFactory;
+import com.google.api.gax.tracing.MetricsTracerFactory;
+import com.google.api.gax.tracing.OpentelemetryMetricsRecorder;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.truth.Truth;
+import com.google.rpc.Status;
+import com.google.showcase.v1beta1.BlockRequest;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoRequest;
+import com.google.showcase.v1beta1.EchoSettings;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import com.google.showcase.v1beta1.stub.EchoStubSettings;
+import io.grpc.ManagedChannelBuilder;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+public class ITOtelMetrics {
+
+  @AfterClass
+  public static void cleanup_otelcol() throws Exception {
+    Process process = Runtime.getRuntime().exec("../scripts/cleanup_otelcol.sh");
+    process.waitFor();
+  }
+
+  @Test
+  public void testHttpJson_OperationSucceded_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory1",
```

Comment:
qq, why do these all need to be a in different directory?

`directory1` vs `directory2`?

Reply:
I tried collecting them in one directory, but that overwrites the metrics of one test into another file. I can make this cleaner once the logic is approved.

Reply:
Now using only one directory.